### PR TITLE
BugFix - Druid Airflow Exception to about content

### LINKED ISSUE #27029: Druid Operator is not getting host
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

 We use Airflow 2.3.3. I see that this test is successful, but I take a this error. This is the picture

```
File "/home/airflow/.local/lib/python3.7/site-packages/requests/sessions.py", line 792, in get_adapter
    raise InvalidSchema(f"No connection adapters were found for {url!r}")
```

Screen Shot 2022-10-12 at 15 34 40

Screen Shot 2022-10-12 at 15 37 27

I take dag like this
Screen Shot 2022-10-13 at 12 36 25
Screen Shot 2022-10-13 at 12 37 15

Also I tried this type but I failed

```python
ingestion_2 = SimpleHttpOperator(
    task_id='test_task',
    method='POST',
    http_conn_id=DRUID_CONN_ID,
    endpoint='/druid/indexer/v1/task',
    data=json.dumps(read_file),
    dag=dag,
    do_xcom_push=True,
    headers={
        'Content-Type': 'application/json'
    },
    response_check=lambda response: response.json()['Status'] == 200)
```

I get this log 
```
[2022-10-13, 06:16:46 UTC] {http.py:143} ERROR - {"error":"Missing type id when trying to resolve subtype of [simple type, class org.apache.druid.indexing.common.task.Task]: missing type id property 'type'\n at [Source: (org.eclipse.jetty.server.HttpInputOverHTTP); line: 1, column: 1]"}
```
I don't know this is bug or issue or networking problem but can we check this?  

P.S - We use Airflow on Kubernetes so that we can not debug it.

### What you think should happen instead

_No response_

### How to reproduce

_No response_

### Operating System

Kubernetes

### Versions of Apache Airflow Providers

_No response_

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by emincanoguz11**
Hi,
This PR is about 
Closes: #27029 

I tested with local env and docker. I see that this result

```python
raise AirflowException(f'The error code of {url}: {req_index.content}')
airflow.exceptions.AirflowException: The error code of DRUID-URL/druid/indexer/v1/task: b'{"error":"Cannot construct instance of `org.apache.druid.indexer.partitions.DimensionRangePartitionsSpec`, problem: partitionDimensions must be specified\\n at [Source: (org.eclipse.jetty.server.HttpInputOverHTTP); line: 102, column: 7] (through reference chain: org.apache.druid.indexing.common.task.batch.parallel.ParallelIndexSupervisorTask[\\"spec\\"]->org.apache.druid.indexing.common.task.batch.parallel.ParallelIndexIngestionSpec[\\"tuningConfig\\"]->org.apache.druid.indexing.common.task.batch.parallel.ParallelIndexTuningConfig[\\"partitionsSpec\\"])"}'
```

P.S : I deleted my own Druid Broker host

Screen Shot 2022-10-20 at 18 04 21

Also I take pictures for standarts

Screen Shot 2022-10-21 at 08 33 31
Screen Shot 2022-10-21 at 08 33 41
Screen Shot 2022-10-21 at 08 33 48
Screen Shot 2022-10-21 at 08 34 09

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

related: #27029 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/apache/druid/hooks/druid.py [hunk 1]
```diff
@@ -89,7 +89,8 @@ def submit_indexing_job(self, json_index_spec: dict[str, Any] | str) -> None:
         self.log.info("Druid ingestion spec: %s", json_index_spec)
         req_index = requests.post(url, data=json_index_spec, headers=self.header, auth=self.get_auth())
         if req_index.status_code != 200:
-            raise AirflowException(f'Did not get 200 when submitting the Druid job to {url}')
+            raise AirflowException(f'The error code of {url}: {req_index.content}')
```
Comment (o-nikolas) [inline airflow/providers/apache/druid/hooks/druid.py]: I find the wording here a little misleading, you're not actually logging a satus code, but rather the response body which contains the error message and traces. Maybe something like:

```suggestion
            raise AirflowException(f'Failed when submitting the Druid job to {url}. Status code: {req_index.status_code}, response: {req_index.content}')
```

**Review by o-nikolas — CHANGES_REQUESTED** [review]

File: airflow/providers/apache/druid/hooks/druid.py (hunk 1 above)
Comment (emincanoguz11) [reply airflow/providers/apache/druid/hooks/druid.py]: Hi Nikolas,

Good advice. I changed that.

**Review by o-nikolas — COMMENTED** [review]
Thanks for making that update! A committer will be required to run the workflow for a first time committer (and ultimately merge if everything passes and they are happy with the code). Unfortunately I am not a committer, I'll CC a few who may have time to have a look: @eladkal @potiuk @kaxil

**Review by uranusjr — APPROVED** [review]

**Comment by o-nikolas** [comment]
@uranusjr has run the build workflow on you PR and there were some static check failures, you'll need to patch those up.
You can enable pre-commit to easily run static checks on your code ([readme](https://github.com/apache/airflow/blob/main/STATIC_CODE_CHECKS.rst#installing-pre-commit-hooks)).

Do also rebase your branch since some changes were recently merged to standardize quoting in the Airflow codebase.

**Comment by emincanoguz11** [comment]
Thanks Nikolas. I am very busy for this weekend. I will check on Friday.

**Comment by o-nikolas** [comment]
> Thanks Nikolas. I am very busy for this weekend. I will check on Friday.

No worries, thanks for sticking with it :)

> P.S @o-nikolas Is this problem about character number or write type? I don't know.
> 
> Screen Shot 2022-10-25 at 21 58 33

This is mypy warning you that at least one of the strings you're formatting here is a byte string, and that Python 3 doesn't handle this edge case gracefully when formatting with f-strings.
You can see from the example output in the screenshot you linked in the description that the logged output contains the `b` prefix: `... b'{"error":...`. calling a `.decode('utf-8')` on the values which are bytstrings will likely solve your issue (assuming they're utf-8 encoded).

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
You need to rebase I am afraid

**Comment by emincanoguz11** [comment]
@potiuk Today, I will rebase and change. Thanks to approved

**Review by emincanoguz11 — COMMENTED** [review]
I changed f string to format type. Can we check this feature?

**Comment by uranusjr** [comment]
I merged main and moved the error message to task log instead, since the HTTP response string can be potentially long and not belong to the exception.

**Review by potiuk — APPROVED** [review]

**Pull request merged**